### PR TITLE
Add uniqueness validation for votes

### LINKED ISSUE #2020: Add validation for user votes
**I'm submitting a ..**
- [ ] Bug Report

**Current behavior:**

A user is able to vote on an event more than once. 

**Expected correct behavior:**

Each user should only vote 1 time for each event.

**Steps to reproduce:**

I saw the bug happening in events.opensuse.org (where I deleted the second vote). I couldn't reproduce. 
The voting process is an ajax call, I am thinking that perhaps it was somehow triggered twice, not necessarily by user's actions.

We should validate against this, so that our database does not have the same person voting on the same proposal.

### PR DESCRIPTION
**Pull request by kaustubh-nair**
**Checklist**

- [x] I have read the [Contribution & Best practices Guide](https://github.com/openSUSE/osem/blob/master/CONTRIBUTING.md).
- [x] My branch is up-to-date with the upstream `master` branch.
- [x] The tests pass locally with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works(if appropriate).
- [x] I have added necessary documentation (if appropriate).

 Fixes #2020 

Prevents duplicate addition of votes for events

### CODE REVIEW

File: spec/models/vote_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe Vote do
```
Comment (Ana06) [inline spec/models/vote_spec.rb]: you may want to check [`validate_uniqueness_of`](https://www.rubydoc.info/github/thoughtbot/shoulda-matchers/master/Shoulda%2FMatchers%2FActiveRecord:validate_uniqueness_of) instead :wink:

File: app/models/vote.rb [hunk 2]
```diff
@@ -6,6 +6,8 @@ class Vote < ApplicationRecord
 
   has_paper_trail ignore: [:updated_at], meta: { conference_id: :conference_id }
 
+  validates :event, uniqueness: { scope: :user }
```
Comment (Ana06) [inline app/models/vote.rb]: I would write this the other way arround:

```ruby
validates :user, uniqueness: { scope: :event }
```

The error message is slightly different. :smile:

File: spec/models/vote_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  let(:user) { create(:user) }
+  let(:event) { create(:event) }
+  let(:vote) { create(:vote, event: event, user: user) }
+
+  describe 'validation' do
+    it { is.expected_to validate_uniqueness_of(:user).scoped_to(:event) }
+
+    it 'has a valid factory' do
```
Comment (kaustubh-nair) [inline spec/models/vote_spec.rb]: Probably unnecessary right now but I let it be because it might be useful later

Comment (Ana06) [reply spec/models/vote_spec.rb]: In what way could it be useful later? i don't think leaving unneeded code is a good idea. We also have to maintain this code. If we need it in the future, we will add it then :wink:

File: app/models/vote.rb [hunk 4]
```diff
@@ -6,6 +6,8 @@ class Vote < ApplicationRecord
 
   has_paper_trail ignore: [:updated_at], meta: { conference_id: :conference_id }
 
+  validates :user_id, uniqueness: { scope: :event_id }
```
Comment (Ana06) [inline app/models/vote.rb]: Why are you using the id instead of the relationship?

File: spec/models/vote_spec.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe Vote do
+  let(:user) { create(:user) }
```
Comment (Ana06) [inline spec/models/vote_spec.rb]: `user` and `event` are not needed. `let(:vote) { create(:vote) }` will create them. (That's why we use factories, to save all this work :wink: )

Comment (differentreality) [reply spec/models/vote_spec.rb]: @kaustubh-nair would you please take care of the review comments, so that we can merge this?

Comment (kaustubh-nair) [reply spec/models/vote_spec.rb]: Oh sorry @differentreality 
I'll make the changes and push again asap

**Comment by Ana06** [comment]
@kaustubh-nair do you need help to address the requested changes?

**Comment by Ana06** [comment]
Closing due to inactivity. @kaustubh-nair in case you address the requested changes reopen it again so that we review it.

**Pull request closed without merging**